### ISSUE TITLE
Changes to accessibility test

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/balloons-and-static-electricity#190]
The following changes were requested for the accessibility visualization

- [ ] Exchange labels for alerts with @terracoda's classification for alerts
- [x] Remove polite updates when alert content changes and there is no polite alert
- [x] Ensure consistency on multiple platforms
- [x] PDOM should remain to right of sim ifram, even on overflow

**Comment by jessegreenberg** [comment]
Windows 10 + IE 11 is missing entire PDOM, though it has the alert content.

**Comment by jessegreenberg** [comment]
It looks like Safari and IE don't support style = '', will need to use removeAttribute nstead.

**Comment by jessegreenberg** [comment]
Tested on Windows 10 + Chrome, Firefox, IE, and OS Yosemite with Chrome, Safari, Firefox.

MacBook Air with Yosemite is missing entries in the parallel DOM.

**Comment by jessegreenberg** [comment]
Those elements are empty in the Parellel DOM as well, so it isn't an issue with the copy or mutation observers.  Our example caught a but :)

**Comment by jessegreenberg** [comment]
The elements are floating in memory, not yet added to the parallel DOM, so updating their text content isn't working.  Need to modify updateDescriptionItem so that it is modifying the elements without using ```document.getElementById```

**Comment by jessegreenberg** [comment]
There were also erroneous uses of inner text.  The above commit fixes this.

**Comment by jessegreenberg** [comment]
> Exchange labels for alerts with @terracoda's classification for alerts

This should be discussed on 11/1/16 accessibility meeting.

Here is the latest version:
http://www.colorado.edu/physics/phet/dev/html/balloons-and-static-electricity/1.2.0-reader.7/acccessibility-test.html

**Comment by jessegreenberg** [comment]
We discussed this in today's meeting, but didn't talk about the specific labels.  @terracoda can you please review http://www.colorado.edu/physics/phet/dev/html/balloons-and-static-electricity/1.2.0-reader.7/acccessibility-test.html and let me know if you think the labels on the aria-live elements should change?

**Comment by terracoda** [comment]
@jessegreenberg, by "label on the aria live elements" do you mean a technical label or a content label? 
When I mentioned that the aria-live descriptions for the balloons had no labels, I meant that the words "Yellow balloon" and Green balloon are not announced when they are grabbed. I think they need to be. I can provide some examples in the morning.
The reason I wanted to clarify about the term "label" as there are three labeling techniques, and the technically the balloons labels are their respective headings. Are we using a an aria-labledby attribute for the aria-live regions?

**Comment by jessegreenberg** [comment]
Sorry I was unclear, I meant these labels in the demonstration, which are not specific to BASE: ```"<assertive update>"```  ```"<polite update>"``` ```"<assertive alert update>"``` and ```"<polite status update>"```.  I think @emily-phet thought these could be replaced by your three labeling techniques.

Do you think that ```"<assertive update>"```  ```"<polite update>"``` ```"<assertive alert update>"``` and ```"<polite status update>"`` in the example in https://github.com/phetsims/balloons-and-static-electricity/issues/190#issuecomment-257708527 should be labelled differently?

> "Yellow balloon" and Green balloon are not announced when they are grabbed. I think they need to be....Are we using a an aria-labledby attribute for the aria-live regions?

We are not, and I agree that we should be.  I will create an issue for this.

**Comment by jessegreenberg** [comment]
Gonna go ahead and close this, we will work on this again when we want to spruce up the demonstration.

**Comment by terracoda** [comment]
@jessegreenberg, sorry, I missed your questions on this issue! I'm not sure the categories directly map to the ARIA labels. My categories are: Static Descriptions, Dynamic Descriptions, and Interactive Alerts.

**Comment by terracoda** [comment]
@jessegreenberg, and I agree, we can close this for now. I need to think about it more.

**Comment by jessegreenberg** [comment]
Oh no worries!  Sounds good, these can be changed any time and we can think about it again once we need to work on this example.

**Comment by emily-phet** [comment]
@terracoda @jessegreenberg Just wanted to make a note that we did provide a link to the visualization of the descriptions in the CSUN paper (which there will be public access to late February). We may not need (or be able to) refine to match have the visual labels in the interface match @terracoda framework descriptions, but we might want to take a look and see if any quick refinements would be useful for the CSUN audience. I'm going to leave this open for a bit in case either of you want to comment, but will close it in a few days if there is no activity.

**Comment by terracoda** [comment]
@emily-phet, the tricky part is that "Static Decriptions" and "Dynamic Descriptions" are in the page's content, and the "Interactive Alerts" appear above the content under the ARIA labels and are fired with different politeness levels. 

Adding the framework labels is a good idea, but I just need to figure out how they interweave in the live example.

**Comment by jessegreenberg** [comment]
Right, 'assertive update', 'polite update', 'assertive alert update', and 'polite status update' are references to the ARIA implementation, but all implement the "Interactive Alerts" in @terracoda's framework.

Everything below those sections them are a mix of "Static Descriptions" "Dynamic Descriptions".

**Comment by emily-phet** [comment]
Sounds like in the interface that links to work describing @terracoda's framework, the alerts should just be labeled as "Interactive Alerts" (perhaps just above all of the labels for type of alert, so all the different types of alerts are seen as subsets of "interactive Alerts"). Then there could be static labels that indicate static descriptions...and we could just have a note somewhere that the rest are dynamic descriptions. Would that work without too much effort?

**Comment by terracoda** [comment]
@emily-phet, yes that makes sense. I just wasn't sure how to add visual labels without cluttering or confusing things. We can easily add the two main labels separating the two areas: 
1. the hidden aspects of the PDOM - Interaction Alerts
2. the browsable aspects of the PDOM - the Static and Dynamic Descriptions

I would like to point out (if it is not already blatantly obvious) that @jessegreenberg has done a lot of work to test and figure what levels of politeness (of the ARIA alerts) work best for implementing the larger category that I call "Interactive Alerts". This more technical side is not yet discussed in any work - we just talk about "ARIA Live Regions" - not the sub categories, and how the sub categories are useful for sims and description.

**Comment by terracoda** [comment]
@jessegreenberg, @emily-phet we could potentially add some color or some formatting that helps identify the the different categories as well. Would you like me to play with that idea?

**Comment by emily-phet** [comment]
@terracoda - I think the goal is to tweak the interface so that those reading the manuscript will be able to map the framework to what they're seeing in the visual description interface...and to do this as efficiently as possible. If you could spend maybe an hour considering the simplest way to convey the information we want, with the most minimal development effort required, that would be helpful.

**Comment by terracoda** [comment]
@emily-phet, ok, will do.

**Comment by terracoda** [comment]
@emily-phet, what do you think of adding some new headings and explanatory text? Alternatively, we can just use headings, or place the explanatory text just below the H1, instead of below the relates sub headings.

![sim-hierarchy-pdom-edits](https://cloud.githubusercontent.com/assets/1902116/21327051/d6683300-c607-11e6-9005-b6144ac4ef7e.png)

**Comment by terracoda** [comment]
@emily-phet and here's a another possibility. The additions here are a bit tidier.
![sim-hierarchy-v02](https://cloud.githubusercontent.com/assets/1902116/21327977/33dbeed8-c60c-11e6-9c22-63ea01247478.png)

**Comment by terracoda** [comment]
@emily-phet, sorry, there are a few typos in the text. I'll put the suggested text in a google doc:
https://docs.google.com/document/d/1xsucsdRs-Oj1NtYnT4XwiQeuvKxsHEA3PRr6LPLfsj0/edit#

**Comment by terracoda** [comment]
@jessegreenberg, sorry, I didn't mention you in the above comments. You are more than welcome to comment as well.

**Comment by emily-phet** [comment]
@terracoda - since the visualization will be coupled with the manuscript, I don't think we need to add more text to describe what's happening. I think we just need to highlight more strongly (and consistently) where the different description types are. Take a look at this mockup. I just use a legend below the sim, and color+font styling to distinguish the different description categories. Not sure if this is straightfoward to implement, but I think it highlights the differences we want to convey.
[Visual_Auditory_Descriptions.pdf](https://github.com/phetsims/balloons-and-static-electricity/files/664345/Visual_Auditory_Descriptions.pdf)

**Comment by terracoda** [comment]
@emily-phet, your example demonstrates what I was thinking when I said:
> we could potentially add some color or some formatting that helps identify the the different categories

Sorry, I read your follow up comment as negative to that idea, i.e., to try something simpler. I see now that I misinterpreted your comment.

**Comment by emily-phet** [comment]
@terracoda No problem, I mostly made the mockup because it was fun for me :) 
@jessegreenberg When you get a chance, can you let me know if my suggestion seems straightforward and quick for implementation?

**Comment by terracoda** [comment]
@emily-phet, I still think we can make it a bit prettier. @jessegreenberg, would adding css classes be enough that we can target the look of the different descriptions and alerts?

**Comment by jessegreenberg** [comment]
@emily-phet, yes the mockup in that pdf looks like it should be pretty quick to implement.

> would adding css classes be enough that we can target the look of the different descriptions and alerts?

Yes, I would think so!

@emily-phet @terracoda should I proceed with changing it to look like https://github.com/phetsims/balloons-and-static-electricity/issues/190#issuecomment-268304644?

**Comment by terracoda** [comment]
@jessegreenberg and @emily-phet, This mock-up should work. The formatting does not rely on color alone to show the differences. This is important. However, with careful use of typography, borders and backgrounds we should be able to make it look more appealing. I can send an adjusted mock-up in 1 or 2 days.

**Comment by jessegreenberg** [comment]
This issue is going to be continued in https://github.com/phetsims/a11y-research/issues/41 since it is a project wide issue.
